cloudbrain/core.go

```diff
@@ -140,6 +145,45 @@ func (c *Core) CreateInstance(ctx context.Context, providerName string, attr Cre
 	}, nil
 }
 
+// RemoveInstance creates an instance in the database and queues off the cloud
+// create job in the background.
+func (c *Core) RemoveInstance(ctx context.Context, attr DeleteInstanceAttributes) (*Instance, error) {
+	inst, err := c.db.GetInstance(attr.InstanceID)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = c.db.RemoveInstance(inst)
+	if err != nil {
+		cbcontext.LoggerFromContext(ctx).WithFields(logrus.Fields{
+			"err": err,
+		}).Error("error deleting instance in database")
+		return nil, err
+	}
+
+	err = c.bb.Enqueue(background.Job{
+		UUID:       uuid.New(),
+		Context:    ctx,
+		Payload:    []byte(attr.InstanceID),
+		Queue:      "remove",
+		MaxRetries: MaxCreateRetries,
+	})
+	if err != nil {
+		cbcontext.LoggerFromContext(ctx).WithFields(logrus.Fields{
```

Comment:
Same with this, I think we should wrap the error we return and rely on whatever calls `RemoveInstance` to log the error.